numba_cuda/numba/cuda/tests/cudapy/test_debuginfo.py

```diff
@@ -784,6 +790,236 @@ def foo(a):
+            result = cuda.device_array(1, dtype=np.float32)
+            foo[1, 1](result, np.pi)
+            result.copy_to_host()
+
+        result_host = math.sin(np.pi) + math.cos(np.pi)
+        self.assertPreciseEqual(result[0], result_host)
+
+        ir_content = foo.inspect_llvm()[foo.signatures[0]]
+        self.assertFileCheckMatches(ir_content, foo.__doc__)
+
+    def test_DILocation_versioned_variables(self):
+        """Tests that DILocation information for versions of variables matches
+        up to their definition site."""
+
+        @cuda.jit(debug=True)
+        def foo(dest, n):
+            """
+            CHECK: define void @{{.+}}foo
+            CHECK: store i64 5, i64* %"c{{.+}} !dbg ![[STORE5:.+]]
+            CHECK: store i64 1, i64* %"c{{.+}} !dbg ![[STORE1:.+]]
+            CHECK: [[STORE5]] = !DILocation(
+            CHECK: [[STORE1]] = !DILocation(
+            """
+            if n:
+                c = 5
+            else:
+                c = 1
+            # prevents inline of return on py310
+            py310_defeat1 = 1  # noqa
+            py310_defeat2 = 2  # noqa
+            py310_defeat3 = 3  # noqa
+            py310_defeat4 = 4  # noqa
+            dest[0] = c
+
+        result = cuda.device_array(1, dtype=np.int32)
+        foo[1, 1](result, 1)
+        result.copy_to_host()
+        self.assertEqual(result[0], 5)
+
+        ir_content = foo.inspect_llvm()[foo.signatures[0]]
+        self.assertFileCheckMatches(ir_content, foo.__doc__)
+
+    def test_debuginfo_asm(self):
+        def foo():
+            pass
+
+        foo_debug = cuda.jit(debug=True)(foo)
+        foo_debug[1, 1]()
+        asm = foo_debug.inspect_asm()[foo_debug.signatures[0]]
+        self.assertFileCheckMatches(
+            asm,
+            """
+            CHECK: .section{{.+}}.debug
+        """,
+        )
+
+        foo_nodebug = cuda.jit(debug=False)(foo)
+        foo_nodebug[1, 1]()
+        asm = foo_nodebug.inspect_asm()[foo_nodebug.signatures[0]]
+        self.assertFileCheckMatches(
+            asm,
+            """
+            CHECK-NOT: .section{{.+}}.debug
+        """,
+        )
+
+    def test_nrt_debuginfo(self):
+        with override_config("CUDA_ENABLE_NRT", True):
+
+            @cuda.jit
+            def f(x):
+                return x[:5]
+
+            @cuda.jit
+            def g():
+                x = np.empty(10, np.int64)
+                f(x)
+
+            g[1, 1]()
+
+            checks = (
+                """
+            CHECK: call void @"NRT_decref"(i8* %"{{.*}}.meminfo.{{[0-9]+}}")
+            CHECK-NOT: !dbg
+            """
+                * 5
+            )
+
+            ir_content = g.inspect_llvm()[g.signatures[0]]
+            self.assertFileCheckMatches(ir_content, checks)
+
+    def test_nrt_debuginfo_refcount(self):
+        with override_config("CUDA_ENABLE_NRT", True):
+
+            @cuda.jit(debug=True)
+            def sink(*x):
```

Comment:
I'm not sure `*args` is properly supported in the CUDA target, so it'd probably be safer to write this as

```suggestion
            def sink(x):
```

With `*x`, some innocuous change might cause this to fail otherwise.